packages/server/app/Models/User.ts

```diff
@@ -47,6 +49,78 @@ export default class User extends BaseModel {
     public mpeRooms: ManyToMany<typeof MpeRoom>;
     ///
 
+    @belongsTo(() => SettingVisibility, {
+        localKey: 'uuid',
+        foreignKey: 'playlistsVisibilitySettingUuid',
+    })
+    public playlistsVisibilitySetting: BelongsTo<typeof SettingVisibility>;
+
+    @column({ columnName: 'playlists_visibility_setting_uuid' })
+    public playlistsVisibilitySettingUuid: string;
+
+    @beforeCreate()
+    public static async assignPlaylistsVisibilitySetting(
+        user: User,
+    ): Promise<void> {
+        if (user.playlistsVisibilitySettingUuid === undefined) {
+            const publicVisibilitySetting =
+                await SettingVisibility.findByOrFail(
+                    'name',
+                    UserSettingVisibility.enum.PUBLIC,
+                );
+
+            user.playlistsVisibilitySettingUuid = publicVisibilitySetting.uuid;
+        }
+    }
+
+    @belongsTo(() => SettingVisibility, {
+        localKey: 'uuid',
+        foreignKey: 'relationsVisibilitySettingUuid',
+    })
+    public relationsVisibilitySetting: BelongsTo<typeof SettingVisibility>;
+
+    @column({ columnName: 'relations_visibility_setting_uuid' })
+    public relationsVisibilitySettingUuid: string;
+
+    @beforeCreate()
+    public static async assignRelationsVisibilitySetting(
+        user: User,
+    ): Promise<void> {
+        if (user.relationsVisibilitySettingUuid === undefined) {
+            const publicVisibilitySetting =
+                await SettingVisibility.findByOrFail(
+                    'name',
+                    UserSettingVisibility.enum.PUBLIC,
+                );
+
+            user.relationsVisibilitySettingUuid = publicVisibilitySetting.uuid;
+        }
+    }
+
+    @belongsTo(() => SettingVisibility, {
+        localKey: 'uuid',
+        foreignKey: 'devicesVisibilitySettingUuid',
+    })
+    public devicesVisibilitySetting: BelongsTo<typeof SettingVisibility>;
+
+    @column({ columnName: 'devices_visibility_setting_uuid' })
+    public devicesVisibilitySettingUuid: string;
+
+    @beforeCreate()
```

Comment:
Unless I'm mistaken, we should not allow the user to manage his devices visibility
I hope this is not a pain to remove for you..

Reply:
I'm working on that!